core-bundle/src/HttpKernel/ModelArgumentResolver.php

```diff
@@ -71,14 +73,11 @@ private function fetchModel(Request $request, ArgumentMetadata $argument): Model
             return $value;
         }
 
+        $pageModel = $this->pageFinder->getCurrentPage();
+
         // Special handling for pageModel that could be globally registered
-        if (
-            isset($GLOBALS['objPage'])
-            && $GLOBALS['objPage'] instanceof PageModel
-            && $GLOBALS['objPage']->id === (int) $value
-            && is_a($type, PageModel::class, true)
-        ) {
-            return $GLOBALS['objPage'];
+        if ($pageModel && $pageModel->id === (int) $value && is_a($type, PageModel::class, true)) {
```

Comment:
This would not load the PageModel if the argumet is not of type PageModel (also needs to delete the line above).

```suggestion
        if (is_a($type, PageModel::class, true) && ($pageModel = $this->pageFinder->getCurrentPage()) && $pageModel->id === (int) $value) {
```

Reply:
Changed in 188c42abe3b2517f5b2285c11276d4f12f8d0807.